src/ILLink.Shared/Dataflow/SingleValue.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace ILLink.Shared
+{
+	// Similar to ValueNode, and a candidate for future code sharing.
+	// This is a sum type over the various kinds of values we track:
+	// - dynamicallyaccessedmembertypes-annotated locations (types or strings)
+	// - known typeof values and similar
+	// - known strings
+	// - known integers
+
+	// The implementation is taken from the generated code for records.
```

Comment:
I'm pretty sure we can just add a definition for System.Runtime.CompilerServices.IsExternalInit into the project, and records will work fine.

Reply:
Neat, thanks! I changed this to be a trivial record :)